pkg/operator/reporting/templates_test.go

```diff
@@ -0,0 +1,525 @@
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery: newTestReportQuery(ds1, "SELECT datetime {| default .Report.ReportingStart .Report.Inputs.ReportingStart | prometheusMetricPartitionFormat |} AS period_start"),
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"ReportingStart": reportStart},
+				},
+			},
+			expectOutput: fmt.Sprintf("SELECT datetime %s AS period_start", reportStart.Format(prestostore.PrometheusMetricTimestampPartitionFormat)),
+		},
+		{
+			name: "query containing valid prestoTimestamp and prometheusMetricPartitionFormat templating results in nil and the expected query output",
+			reportTemplate: &ReportQueryTemplateContext{
+				ReportQuery: newTestReportQuery(ds1, "SELECT timestamp {| default .Report.ReportingStart .Report.Inputs.ReportingStart | prestoTimestamp |} AS period_start WHERE dt >= {| default .Report.ReportingStart .Report.Inputs.ReportingStart | prometheusMetricPartitionFormat |}"),
+			},
+			templateContext: TemplateContext{
+				Report: ReportTemplateInfo{
+					Inputs: map[string]interface{}{"ReportingStart": reportStart},
+				},
+			},
+			expectOutput: fmt.Sprintf("SELECT timestamp %s AS period_start WHERE dt >= %s", reportStart.Format(presto.TimestampFormat), reportStart.Format(prestostore.PrometheusMetricTimestampPartitionFormat)),
+		},
+	}
+
+	for _, testCase := range testTable {
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			output, err := RenderQuery(testCase.reportTemplate, testCase.templateContext)
+
+			if testCase.expectErr {
+				assert.EqualErrorf(t, err, testCase.expectErrMsg, "expected that RenderQuery would return the correct error message")
+			} else {
+				assert.NoErrorf(t, err, "expected the report would return no error, but got an error.")
+				assert.EqualValuesf(t, testCase.expectOutput, output, "expected that RenderQuery would return the expected output")
+			}
+		})
+	}
+}
+
+// newPrestoTableCustom allows us to insert a custom name, as opposed to testhelpers.NewPrestoTable
+func newTestPrestoTable(name, namespace, schema string, columns []presto.Column) *v1alpha1.PrestoTable {
+	return &v1alpha1.PrestoTable{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Status: v1alpha1.PrestoTableStatus{
+			Catalog:   testCatalogName,
+			Schema:    schema,
+			TableName: name,
+			Columns:   columns,
+		},
+		Spec: v1alpha1.PrestoTableSpec{
+			Catalog:   testCatalogName,
+			Schema:    schema,
+			TableName: name,
+			Columns:   columns,
+		},
+	}
+}
+
+// newReportQueryCustomQuery helps create a ReportQuery with a custom ReportDataSource and adds a Query field
+func newTestReportQuery(dataSource *metering.ReportDataSource, query string) *metering.ReportQuery {
+	return &metering.ReportQuery{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testReportName,
```

Comment:
Nit: globals are generally speaking, not good, so maybe these should be arguments to the function